x/nexus/keeper/general_message.go

```diff
@@ -10,26 +10,24 @@ import (
 	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
 )
 
-func getMessageKey(id exported.MessageID) key.Key {
-	return generalMessagePrefix.
-		Append(key.From(id.Chain)).
-		Append(key.FromStr(id.ID))
+func getMessageKey(id string) key.Key {
+	return generalMessagePrefix.Append(key.FromStr(id))
 }
 
 // SetNewMessage sets the given general message
 func (k Keeper) SetNewMessage(ctx sdk.Context, m exported.GeneralMessage) error {
-	sourceChain, ok := k.GetChain(ctx, m.SourceChain)
+	sourceChain, ok := k.GetChain(ctx, m.GetSourceChain())
 	if !ok {
-		return fmt.Errorf("source chain %s is not a registered chain", m.SourceChain)
+		return fmt.Errorf("source chain %s is not a registered chain", m.GetSourceChain())
 	}
 
-	destChain, ok := k.GetChain(ctx, m.ID.Chain)
+	destChain, ok := k.GetChain(ctx, m.GetDestinationChain())
 	if !ok {
-		return fmt.Errorf("destination chain %s is not a registered chain", m.ID.Chain)
+		return fmt.Errorf("destination chain %s is not a registered chain", m.GetDestinationChain())
 	}
 
 	validator := k.GetRouter().GetAddressValidator(destChain.Module)
-	if err := validator(ctx, exported.CrossChainAddress{Chain: destChain, Address: m.Receiver}); err != nil {
+	if err := validator(ctx, m.Recipient); err != nil {
```

Comment:
should we also validate `m.Sender` as well?

Reply:
it is validated before, but yes, I think still good to validate sender here